packages/dev/core/src/Rendering/objectRenderer.ts

```diff
@@ -509,14 +532,22 @@ export class ObjectRenderer {
 
         this._currentSceneCamera = this._scene.activeCamera;
 
+        if (this._useUBO) {

```

Comment:
I am not sure to fully get why we need the swap here ?

Reply:
The `ObjectRenderer` needs its own scene ubo to make the fast snapshot rendering mode work as expected (hard to explain with words, we can discuss it offline if you want).